infrastructure/modules/ansible/roles/nimona_bootstrap/tasks/main.yml

```
@@ -34,6 +34,8 @@
         {{ nimona_bootstrap_private_keys[inventory_hostname] }}
       NIMONA_METRICS_BIND_ADDRESS: >-
         0.0.0.0:{{ nimona_bootstrap_metrics_port }}
+      NIMONA_METRICS_PYROSCOPE_SERVER_URL: >-
+        "https://metrics.staging.nimona.dev/pyroscope/"
```

I believe this is what you want to avoid the hard-coded hostname :)

```suggestion
      NIMONA_METRICS_PYROSCOPE_SERVER_URL: >-
        "https://{{ inventory_hostname }}/pyroscope/"
```